clang/lib/CodeGen/CodeGenTBAA.cpp

```diff
@@ -80,6 +80,28 @@ llvm::MDNode *CodeGenTBAA::getChar() {
   return Char;
 }
 
+llvm::MDNode *CodeGenTBAA::getAnyPtr(unsigned PtrDepth) {
+  assert(PtrDepth >= 1 && "Pointer must have some depth");
+
+  // Populate at least PtrDepth elements in AnyPtrs. These are the type nodes
+  // for "any" pointers of increasing pointer depth, and are organized in the
+  // hierarchy: any pointer <- any p2 pointer <- any p3 pointer <- ...
```

Comment:
```suggestion
  // hierarchy: any pointer > any p2 pointer > any p3 pointer > ...
  // These "any" types are used for void pointers, as well as some
  // cases where we're just being conservative. Essentially, the rule
  // we're using is that void** ought to be able to alias any type that
  // can unify with T** (where T is a variable), ignoring things like
  // qualifiers. This naturally makes each depth a subtype of the
  // previous depth, and thus transitively of all previous depths.
  //
  // Note that element idx of AnyPtrs is the node for pointer depth
  // (idx+1), since there's no node for pointer depth 0.
```

I'm suggesting `>` here rather than arrows because I think it expresses the subtype relationship more clearly, but I might be abnormally comfortable with subtyping lattices, so let me know what you think.

Reply:
I've extended the comment based on your suggestion, let me know what you think!

> I'm suggesting `>` here rather than arrows because I think it expresses the subtype relationship more clearly, but I might be abnormally comfortable with subtyping lattices, so let me know what you think.

Yeah, I did not find this very clear. I used arrows to express the parent relationships. I don't think it is common to describe TBAA type nodes with subtyping, whereas "parent" is the term used in the documentation and API? I can still adapt this if you or others prefer[^1].

[^1]: I will say that I'm more used to the symbol `<:` for the subtype relationship, e.g.:
![image](https://github.com/user-attachments/assets/1202957b-133a-4656-a515-97a718792aaa)